### PR TITLE
feat: dc_onlyfans_fansdb Fansly support

### PR DESCRIPTION
**Pull request by xantror**
This PR adds support for scraping Fansly metadata from `user_data.db` files and improves upon a few other things.

### CODE REVIEW

File: scrapers/dc_onlyfans_fansdb/dc_onlyfans_fansdb.py [hunk 1]
```diff
@@ -270,53 +258,68 @@ def find_gallery_path(gallery_id):
...
+
+def get_studio_info(studio_name, studio_network):
     """
-    Find and return OnlyFans (network) `stored_id` else return only name
+    Resolve studio based on name and network
     """
-    studios = stash.find_studios(
-        f={"name": {"value": "OnlyFans (network)", "modifier": "EQUALS"}},
+    req = stash.find_studios(
+        f={"name": {"value": f"{studio_name} ({studio_network})", "modifier": "EQUALS"}},
```
Comment (Maista6969) [inline scrapers/dc_onlyfans_fansdb/dc_onlyfans_fansdb.py:293]: You could make the same change as I suggested for performers here if you'd like to support people having their studios aliased

You could even make the query more complex and find the studio with the `studio_name` as its name or alias AND the parent studio called `studio_network`, but that's probably overcomplicating things

File: scrapers/dc_onlyfans_fansdb/dc_onlyfans_fansdb.py [hunk 2]
```diff
@@ -344,6 +351,11 @@ def find_performer_images(path, max_images=MAX_PERFORMER_IMAGES):
             [CACHE MISS] Encoding {index + 1} of {len(selected_images)} image(s) to base64: {image}'
         """)
         base64_data = file_to_base64(image)
+        if base64_data is None:
+            log.error("Error converting image to base64: {image}")
```
Comment (Maista6969) [inline scrapers/dc_onlyfans_fansdb/dc_onlyfans_fansdb.py:355]: This should also be an f-string

File: scrapers/dc_onlyfans_fansdb/dc_onlyfans_fansdb.py [hunk 3]
```diff
@@ -402,78 +415,84 @@ def format_title(title, username, date, scene_index):
...
-        if index + 1 < len(path):
-            return path[index + 1]
-        else:
-            return None  # Element found but no item follows it
+        index = parts.index(network)
+        if index + 1 < len(parts):
+            return parts[index + 1], network, Path(*parts[:index + 2])
+
+        raise ValueError
     except ValueError:
```
Comment (Maista6969) [inline scrapers/dc_onlyfans_fansdb/dc_onlyfans_fansdb.py:464]: There really is no point in raising a ValueError if you're going to catch it on the very next line 😁 Just drop these two lines

**Review by Maista6969 — APPROVED** [review]
Looks good, all I found were nits! Let me know if you want me to merge this without any further changes 👍

**Comment by xantror** [comment]
Fixed the f-strings and implemented the performer/studio suggestions.

Happy for merging now.

**Pull request merged**